data/items.ts

```diff
@@ -1028,7 +1040,7 @@ export const Items: {[itemid: string]: ItemData} = {
 		onFractionalPriorityPriority: -2,
 		onFractionalPriority(priority, pokemon) {
 			if (
-				priority <= 0 &&
+				priority <= 0 && this.effect.id !== 'confusion' &&
```

Comment:
It's especially unnecessary here, where damage is not the triggering factor in activation.